network/transport/grpc/connection_manager.go

```diff
@@ -387,21 +381,21 @@ func (s *grpcConnectionManager) openOutboundStream(connection Connection, protoc
 	return clientStream, nil
 }
 
-func (s *grpcConnectionManager) authenticate(nodeDID did.DID, peer transport.Peer, peerFromCtx *grpcPeer.Peer) (transport.Peer, error) {
+func (s *grpcConnectionManager) authenticate(nodeDID did.DID, peer transport.Peer, peerFromCtx *grpcPeer.Peer) transport.Peer {
 	if !nodeDID.Empty() {
-		authenticatedPeer, err := s.authenticator.Authenticate(nodeDID, *peerFromCtx, peer)
+		var err error
+		peer, err = s.authenticator.Authenticate(nodeDID, *peerFromCtx, peer)
 		if err != nil {
 			log.Logger().
 				WithError(err).
 				WithFields(peer.ToFields()).
 				WithField(core.LogFieldDID, nodeDID).
 				Warn("Peer node DID could not be authenticated")
 			// Error message is spec'd by RFC017, because it is returned to the peer
-			return transport.Peer{}, ErrNodeDIDAuthFailed
+			//return transport.Peer{}, ErrNodeDIDAuthFailed // TODO: removing this requires a spec change
```

Comment:
do we alter the spec in this change? Since this effectively removes it.

Reply:
Yes, this change should be backwards compatible though. https://github.com/nuts-foundation/nuts-specification/issues/252